xml_schema/tests/choice.rs

```diff
@@ -0,0 +1,101 @@
+extern crate yaserde_derive;
+
+use xml_schema_derive::XmlSchema;
+use yaserde::de::from_str;
+use yaserde::ser::to_string;
+
+#[test]
+fn choice() {
+  #[derive(Debug, XmlSchema)]
+  #[xml_schema(source = "xml_schema/tests/choice.xsd")]
+  struct ChoiceTypeSchema;
+
+  use choice_type_schema::xml_schema_types::*;
+
+  let xml_1 = r#"<?xml version="1.0" encoding="UTF-8"?>
+  <person>
+    <firstname>John</firstname>
+  </person>
+  "#;
+
+  let sample_1: Person = from_str(xml_1).unwrap();
+
+  let model = Person {
+    firstname: Some(Firstname {
+      base: "John".to_string(),
+      scope: None,
+    }),
+    lastname: None,
```

Comment:
In that case I expect more to retrieve an enum here.
The test is not easy to understand as a person has basically a first name and a last name, so it's not a choice, more a combination.

I think the example here is more relevant: https://www.w3schools.com/xml/el_choice.asp
Where. Person may have role `employee` or `member`.

So for me the expected code will be:
```rust
let model = Person::Employee("John".to_string());

// Generated part by xml-schema
enum Person {
  Employee(String),
  Member(String),
}
```